lib/resty/jwt.lua

```diff
@@ -549,7 +551,7 @@ function _M.sign(self, secret_key, jwt_obj)
     end
   end
 
-  if typ == str_const.JWE or jwt_obj.header.enc then
+  if jwt_obj.typ == str_const.JWE or (jwt_obj.typ == nil and (typ == str_const.JWE or jwt_obj.header.enc)) then
     return sign_jwe(self, secret_key, jwt_obj)
   end
```

Comment:
The way to check if the obj is jwe is not correct. The jwe obj has five dot in the body, i think it's more appropriate

Reply:
And the function `sign_jwe` is not correct too, you should refactor the function